core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -629,12 +630,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       AdminUtils.createTopic(zkUtils, topic, numPartitions, replicationFactor, properties, RackAwareMode.Safe)
       info("Auto creation of topic %s with %d partitions and replication factor %d is successful"
         .format(topic, numPartitions, replicationFactor))
-      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, java.util.Collections.emptyList())
+      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, common.Topic.isInternal(topic),
```

Comment:
This is a bit different. There are 3 levels, in a sense:

1. Wire protocol
2. Wrapping the wire protocol into domain model classes
3. Having business logic like what topics are internal in the request classes

As we discussed previous @granthenke, I think we should really do `1`, but we do `2` in a number of places. This suggestion is `3`. In any case, I'm OK if we do it in a static factory method as a starting point (and we can move it elsewhere in the future in a separate PR maybe).